### PR TITLE
Preserve commit author when importing patches.

### LINKED ISSUE #4: Warn when a Phabricator patch has no commits
We should warn when https://github.com/mozilla/libmozevent/blob/68e44aa4eb7272bb3db920611d39222f761d9b77/libmozevent/mercurial.py#L147 is falsy.

### PR DESCRIPTION
**Pull request by La0**
Fixes https://github.com/mozilla/code-review/issues/101 and fixes #4

### CODE REVIEW

File: tests/mocks/phabricator/search-PHID-DIFF-test123.json [hunk 1]
```diff
@@ -29,7 +29,11 @@
                   "commits": {
                     "commits": [
                       {
-                        "message": "Bug XXX - A second commit message"
+                        "message": "Bug XXX - A second commit message",
+                        "author": {
+                          "name": "John Doe",
+                          "raw": "John Doe <john@allizom.org>"
```
Comment (sylvestre) [inline tests/mocks/phabricator/search-PHID-DIFF-test123.json]: Maybe add a test with an invalid email address?
From the code, I don't think it has an impact currently but it might in the future?!

**Comment by La0** [comment]
Test revision with author preserved: https://treeherder.mozilla.org/#/jobs?repo=nss-try&revision=e817f95d79a240635dd54afab43c23f4596786ce

File: libmozevent/mercurial.py [hunk 2]
```diff
@@ -189,7 +207,7 @@ def add_try_commit(self, build):
             }
             message = "try: {}".format(self.try_syntax)
             if build.revision_url is not None:
-                message += f"Phabricator Revision: {build.revision_url}"
+                message += f"\nPhabricator Revision: {build.revision_url}"
```
Comment (marco-c) [inline libmozevent/mercurial.py]: Why this change?

Comment (La0) [reply libmozevent/mercurial.py]: Because i'm dumb, and [made this](https://treeherder.mozilla.org/#/jobs?repo=nss-try&revision=f75ea700952ae28cb2547b5e8b90f5da15b5397e).
